### ISSUE TITLE
poor handling of LaTeX in x- and y-axis labels

### ISSUE THREAD

**Issue by SimonEnsemble** [JuliaPlots/Makie.jl#2197]
:heart: Makie generally, but to include these plots in scientific publications, *imperative* for me to include math in my x- and y-axis labels.

e.g. there seems to be no "q" subscript in Unicode so I must resort to LaTeX here. and it gives me two different fonts for my x- and y-axis labels. any way to mix LaTeX font for the variables with regular (nice-looking) font for the non-math mode? see how the x- and y-axis labels are in a different font?

sure, a solution for consistent font is to use LaTeX  mode for everything, but this font for LaTeX is incredibly ugly. I'm thinking about `matplotlib` that allows LaTeX and text seamlessly. 

```
ax  = Axis(fig[2, 1], xlabel=L"$c_{eq}$ [ng/cm$^3$]", ylabel="D [nm²/s]")
```

![image](https://user-images.githubusercontent.com/30643894/182737893-cae09ca7-72f6-4d66-ae25-0588578d28e7.png)

**Comment by lazarusA** [comment]
I use "CMU Serif", it's not exactly the same but it looks good enough.
https://lazarusa.github.io/BeautifulMakie/ScattersLines/besselFunctions/
![besselFunctions](https://user-images.githubusercontent.com/19525261/182792240-ae0c960a-6ce0-4309-9519-c98f48795b41.svg)

**Comment by jkrumbiegel** [comment]
I also thought that it might be worthwhile to add some really simple markup syntax to our default layout algorithm which could handle super and subscripts. It's kind of unnecessary to rely on unicode for that, especially because we know coverage is not great. Maybe something html-looking like `x<sub>q</sub>` or so?

**Comment by SimonEnsemble** [comment]
thanks for the tip! this is a decent patch, but honestly not a fan of the CMU Serif font / being restricted to this for all of my plots.

allowing for html to make super and subscripts would be a great start! I'm finding myself changing variable names in my papers (to less ideal/clear ones) to accommodate Unicode/CairoMakie. :eyes:

**Comment by SimonEnsemble** [comment]
@jkrumbiegel is there any way to make a subscript \infty? :cry:

**Comment by lazarusA** [comment]
you mean this, or?

```Julia
using GLMakie, LaTeXStrings

fig, ax, obj = lines(0..2π, sin;
    label = L"sin_{\infty}(x)",
    figure = (;fontsize = 38))
axislegend(ax)
fig
```
<img width="394" alt="Screenshot 2022-09-25 at 18 31 59" src="https://user-images.githubusercontent.com/19525261/192154590-b52aade7-8977-47c3-b734-3a1ff7324bbe.png">

**Comment by SimonEnsemble** [comment]
@lazarus, I'm trying to avoid the ugly LaTeX font so was wondering if this is possible with the unicode character. 
maybe I'm better off asking `MathTeXEngine.jl` to implement a sans serif LaTeX font?

**Comment by SimonDanisch** [comment]
This could be done with `RichText` now, if it wasn't for https://github.com/MakieOrg/Makie.jl/issues/2838

```julia
fig, ax, obj = lines(0 .. 2π, sin;
figure=(; fontsize=38),
axis=(; title=rich("sin", subscript("∞"), "(x)")))
fig
```
<img width="399" alt="image" src="https://user-images.githubusercontent.com/1010467/234843408-8957cba8-af5e-4fff-9536-2ce7f762c6d1.png">

**Comment by SimonEnsemble** [comment]
yeah, the `rich` text = a huge improvement. been using it. thanks!

that said, something like supporting fira math font or allowing to change the LaTeX style would still be a game changer for me. eg. snapshot from my recent paper (barely acceptable).


<img width="470" alt="Screenshot 2023-04-27 at 10 53 52 AM" src="https://user-images.githubusercontent.com/30643894/234951498-e4fbb61e-3a56-4513-b7d4-92ea7fc6ecbc.png">

**Comment by SimonDanisch** [comment]
@Kolaru pointed out that one just needs 2 small little tweaks per font, which are probably these numbers here:
https://github.com/Kolaru/MathTeXEngine.jl/blob/master/src/engine/fonts.jl#L91-L106
Would be nice to have a little test plot, to see & tweak those numbers and make it easy to add a new font this way

**Comment by jkrumbiegel** [comment]
I also suggested a little GLMakie app to tweak these per font.

And I had super-subscript for `rich` before, but removed it due linebreak ambiguities. But I might be able to bring it back
